### DISCUSSION TITLE
Book mode flippingPage

### DISCUSSION THREAD

**Discussion by ibtissemoussaifi** [stephanrauh/ngx-extended-pdf-viewer#1435] [Q&A]
Hello,  
can you help me please ?
- I want to change the first and the last page flipping like the others, I try changing 'hard' to 'soft' in viewer.js
and changing drawHard() function with drawSoft() function, I get the result with a little problem whitch is:
when I am in the second page and want to flip to the first page, So, the problem is the first page rest seen to the end, but I want it to be hidden from the first second.
- Another question please, can I use pdf with portrait and landscape page at the same document.
Thanks.
Best Regards

**Comment by stephanrauh** [comment]
That's two questions:

- Soft cover for the first and last page: that makes sense to me. I'm not 100% sure I understand the bug you're reporting, but I've seen similar weird effects, so I guess I know where you're heading. In any case, replacing the hard cover by a soft cover is probably a piece of cake to me, but a huge challenge to you (unless you want to provide a pull request to https://github.com/stephanrauh/pdf.js, which is always an option!). You can help me by opening a feature request. Such a feature request helps me not to forget your issue.

- PDFs with portrait and landscape: well, yeah, that's already possible. One of the demos of the showcase uses such a file: https://pdfviewer.net/extended-pdf-viewer/mobile. But you're mentioning it, so you've probably run into trouble when you tried. Would you mind to elaborate?

**Comment by ibtissemoussaifi** [comment]
- For the second question, I want to use PDF with portrait and landscape in book mode , it causes a problem that the second page which is landscape is cropped as you see here [https://github.com/ibtissemoussaifi/PFE/blob/master/capture.PNG]

**Reply by stephanrauh** [reply]
I see. I forgot about the book mode. I'm afraid it's difficult to implement this feature in book mode. You can try this yourself. The book mode project is here: https://nodlik.github.io/StPageFlip/. Try to configure the StPageFlip library so it can deal with both horizontal and vertical images. If you've figured out how to do it, show me the solution, and I'll try to port it to ngx-extended-pdf-viewer.

**Reply by ibtissemoussaifi** [reply]
Thanks for that , but  I think that I can't use this package [ https://nodlik.github.io/StPageFlip/] 
because I worked with a pdf file and not by images.

**Reply by stephanrauh** [reply]
I know that. :) My idea was completely different. At the moment, I don't have enough spare time to develop such a complex feature. But if I can convince you to lay some groundwork, I'm ready to pick up your prototype and add it to ngx-extended-pdf-viewer. 

BTW, the reason why I'm talking about images is simple: the PDF viewer renders every page as an image.

**Reply by ibtissemoussaifi** [reply]
Hello, sorry for the late response :( , I'm trying to find a solution and I will send it :)

**Reply by stephanrauh** [reply]
That's very kind of you! Thanks you in advance - no matter if you find a solution or not, I appreciate your offer!

**Reply by ibtissemoussaifi** [reply]
Hello,
I resolve it  :) 
Using pageFlip, i find a solution to use PDF with portrait and landscape in book mode

Regards

**Reply by stephanrauh** [reply]
Is this related to #1497, or is it a different issue?

**Reply by ibtissemoussaifi** [reply]
It is related to it
